src/store/analyze.rs

```diff
@@ -59,46 +63,41 @@ impl<'a> fmt::Debug for Match<'a> {
     }
 }
 
-// this could probably be a stand-alone closure, but I was hitting lifetime
-// hell, so a macro it is. feel free to attempt it yourself.
-macro_rules! analyze_fold_closure {
-    ($text:ident) => {
-        |mut acc: Vec<PartialMatch<'_>>, (name, data)| {
-            acc.push(PartialMatch {
-                score: data.original.match_score($text),
-                name,
-                license_type: LicenseType::Original,
-                data: &data.original,
-            });
-            data.alternates.iter().for_each(|alt| {
-                acc.push(PartialMatch {
-                    score: alt.match_score($text),
-                    name,
-                    license_type: LicenseType::Alternate,
-                    data: alt,
-                })
-            });
-            data.headers.iter().for_each(|head| {
-                acc.push(PartialMatch {
-                    score: head.match_score($text),
-                    name,
-                    license_type: LicenseType::Header,
-                    data: head,
-                })
-            });
-            acc
-        }
-    };
-}
-
 impl Store {
     /// Compare the given `TextData` against all licenses in the `Store`.
     ///
     /// This parallelizes the search as much as it can to find the best match.
     /// Once a match is obtained, it can be optimized further; see methods on
     /// `TextData` for more information.
-    pub fn analyze(&self, text: &TextData) -> Match<'_> {
-        let mut res: Vec<PartialMatch<'_>>;
+    pub fn analyze<'a>(&'a self, text: &TextData) -> Match<'a> {
+        let mut res: Vec<PartialMatch<'a>>;
+
+        let analyze_fold =
```

Comment:
Awesome work on getting that into an actual closure instead of a macro. 🎉

Reply:
Haha thanks, I guess I took the comment as a challenge.